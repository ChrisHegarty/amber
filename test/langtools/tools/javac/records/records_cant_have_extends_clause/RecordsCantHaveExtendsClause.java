/*
 * @test /nodynamiccopyright/
 * @summary check that a datum can inherit from DataClass or an abstract datum class
 * @compile/fail/ref=RecordsCantHaveExtendsClause.out -XDrawDiagnostics RecordsCantHaveExtendsClause.java
 */

import java.io.Serializable;

class RecordsCantHaveExtendsClause {

    // even Object which is the implicit super class for records
    record R1(int x) extends Object {}

    record R2(int y) {}

    // can't extend other records either
    record R3(int y, int x) extends R2(y) {}

    // records can implement interfaces
    record R4() implements Serializable, Runnable {
        public void run() {}
    }
}
