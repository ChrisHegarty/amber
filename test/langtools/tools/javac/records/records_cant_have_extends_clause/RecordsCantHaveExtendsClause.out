RecordsCantHaveExtendsClause.java:12:21: compiler.err.expected: '{'
RecordsCantHaveExtendsClause.java:17:28: compiler.err.expected: '{'
2 errors
